### ISSUE TITLE
Running xonsh fails when calling functions from the inspect module

### ISSUE THREAD

**Issue by mardukbp** [indygreg/PyOxidizer#245]
[xonsh](https://github.com/xonsh/xonsh) is a cross-platform shell written in Python and in my opinion it would be great to distribute it as a single executable.

PyOxidizer successfully builds xonsh. However, executing `main()` throws some errors. After fixing the famous issue with `__file__` the remaining issue is that the functions from the `inspect` module, which is part of Python, fail to retrieve the source code of the modules in xonsh. The error message states that inspect considers them built-in modules. I guess this means that the modules have been compiled to machine code, but I could be wrong.

How can one get the inspect module to work in PyOxidizer? I understand that, in principle, compiling code prevents introspection tools from working. But I thought that since PyOxidizer bundles the Python executable, any Python module should be able to work.

**Comment by indygreg** [comment]
Which `inspect` functions or functionality isn't working, @mardukbp?

**Comment by mardukbp** [comment]
Never mind. Last week I got more experienced with PyOxidizer and now I don't have that problem anymore. I set `filesystem_importer=True`, `ignore_environment=False` and `isolated=False` to get past that error on Windows 10 using the Python build that requires shared libs. Now I have a problem with a xonsh module, because after the oxidation it became a built-in module. I will investigate that and if I can't solve it I will open an issue.
